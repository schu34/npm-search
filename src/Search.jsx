import React from "react";
import PropTypes from "prop-types";
import { useInput } from "ink";
import TextInput from "ink-text-input";

const Search = ({ value, onChange, getResults }) => {
  useInput((_, key) => {
    if (key.return) getResults();
  });

  return (
    <TextInput
      placeholder={"search for a package"}
      value={value}
      onChange={onChange}
    />
  );
};

Search.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  getResults: PropTypes.func.isRequired,
};
export default Search;
