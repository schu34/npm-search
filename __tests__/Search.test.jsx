import React from "react";
import Search from "../src/Search";
import { render } from "ink-testing-library";

describe("<Search />", () => {
  it("should render without errors", () => {
    const props = {
      value: "",
      onChange: () => {},
      getResults: () => {}
    };
    const { lastFrame } = render(<Search {...props} />);
    expect(lastFrame()).toMatch("search for a package");
  });
});
