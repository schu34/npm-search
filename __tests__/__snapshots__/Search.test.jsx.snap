// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render without errors 1`] = `"[2msearch for a package[22m"`;
